### PR TITLE
feat (styles): introduce theme driven page templates

### LINKED ISSUE #5159: Layout components styles does not work in Calydon
@developpeurweb @mazamcodes and I took a look at the problem. The components in the Layout folder do not work within the Calydon theme. We found that the component selectors in the the Layout folder are not in the "_sparta-theme.scss".

One possible solution would be to add the Layout component selectors to "_sparta-theme.scss".

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
Similar to the theming we use for components (i.e. calydon), we can now use a theme for a page layouts. The page layout can have addition placeholder selectors, so that whenever the theme is used, additional styles rules are added to the final CSS. 

Additional, we introduced a white and blacklist of page templates so that customers can more easily opt-out of certain page template styles. The variables are `$page-template-blacklist` and `$page-template-whitelist`. 

Moreover, the page templates can now be imported separately, using `@import '@spartacus/styles/scss/page-template';`

closes #5159

### CODE REVIEW

File: projects/storefrontstyles/scss/cxbase/_mixins.scss [hunk 1]
```diff
@@ -1,4 +1,13 @@
+/** 
+ * the mixins depend on variables, mainly because of some
+ * lower level bootstrap variables. Hence we variables here,
```
Comment (LTiger14) [inline projects/storefrontstyles/scss/cxbase/_mixins.scss]: Hence we "import" variables here,

File: projects/storefrontstyles/scss/cxbase/_mixins.scss [hunk 2]
```diff
@@ -1,4 +1,13 @@
...
+ * to make the mixins resuable. In order to have the right variables
+ * we load variables throug the theme import.
```
Comment (LTiger14) [inline projects/storefrontstyles/scss/cxbase/_mixins.scss]: throug -> through

File: projects/storefrontstyles/scss/cxbase/_variables.scss [hunk 3]
```diff
@@ -12,10 +12,13 @@
   so a variable was created to make this available for other themes.
 
 */
+
 @import 'functions';
-@import 'mixins';
 @import '~bootstrap/scss/variables';
 
+// future theme's, can be introduced during minors
```
Comment (LTiger14) [inline projects/storefrontstyles/scss/cxbase/_variables.scss]: future themes, can be introduced during minors

File: projects/storefrontstyles/scss/_page-template.scss [hunk 4]
```diff
@@ -0,0 +1,24 @@
+@import './cxbase/mixins';
+
+@import './layout/index';
+
+// The $page-template-blacklist can be used to prevent template CSS
+// being generated.
+$page-template-blacklist: () !default;
```
Comment (LTiger14) [inline projects/storefrontstyles/scss/_page-template.scss]: Maybe we could rename these two variables. So like, $page-template-names. $page-templates-ignored?

Comment (tobi-or-not-tobi) [reply projects/storefrontstyles/scss/_page-template.scss]: we agreed to keep using black vs whitelists.

**Comment by tobi-or-not-tobi** [comment]
QA steps:
- introduce a page layout CSS rule for `calydon` theme
- verify that it will only be used when the theme is used

**Pull request merged**